Libraries/MiscDrivers/Display/fonts/fonts.mk

```diff
@@ -0,0 +1,40 @@
+###############################################################################
+ #
+ # Copyright 2023 Analog Devices, Inc.
+ #
+ # Licensed under the Apache License, Version 2.0 (the "License");
+ # you may not use this file except in compliance with the License.
+ # You may obtain a copy of the License at
+ #
+ #     http://www.apache.org/licenses/LICENSE-2.0
+ #
+ # Unless required by applicable law or agreed to in writing, software
+ # distributed under the License is distributed on an "AS IS" BASIS,
+ # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+ #
+ ##############################################################################
+# If FONTS_DIR is not specified, this Makefile will locate itself.
+FONTS_DIR ?= $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
```

Comment:
Makefile noob here. Can we always guarantee that fonts.mk will be the last Makefile in the list?

Reply:
Yes, Makefiles are parsed in 2 stages.  In the first stage, it is parsed sequentially through in order.  `include fonts.mk` immediately enter this file and the `FONTS_DIR` variable is created to point to the file's parent directory.

In the second stage, the target rule is run.  This back-propagates through the first run, and recursively resolves any variables that the target depends on.